zapisy/apps/enrollment/records/models/opening_times.py

```diff
@@ -86,18 +101,21 @@ def populate_t0(cls, semester: Semester, students: Iterable[Student] = None):
                 ], student__in=students).values('student_id').annotate(num_tickets=models.Count('id')).values_list(
                     'student_id', 'num_tickets'))
 
+        ranking = cls.get_ects_ranking()
+
         student: Student
         for student in students:
+            # The ECTS bonus is now based on the position in the ECTS
+            # ranking list. Students with the same amount of ECTS have
+            # the same position in the ranking. Each subsequent position
+            # in the ranking gives an additional 2 minutes (by default) of bonus.
+            ects_bonus = semester.get_records_spacing() * ranking[student.ects]
```

Comment:
```suggestion
            ects_bonus = semester.get_records_spacing() * ranking.get(student.ects, 0)
```
Na ten moment interfejs pozwala na obliczenie czasu otwarcia dla osoby nieaktywnej, co w przypadku, w którym ma ona liczbę ECTS niereprezentowaną w rankingu, skutkuje niezłapanym wyjątkiem. Powyższe to właściwie _hotfix_ i być może zmienimy coś jeszcze w logice, jak dowiem się, jak, ale na razie niech będzie przynajmniej coś takiego).

Reply:
Nie wiem, jak szybko uda mi się uzyskać tę informację zwrotną, więc na razie może zostać ww. zmiana, ale – na wypadek, gdyby przyszło _merge_'ować rozwiązanie bez rozwikłania tego do końca – dobrze byłoby, żeby cała operacja wykonywała się z ostrzeżeniem (wyświetlającym się standardowo w panelu administracyjnym), że wśród wybranych studentów były osoby nieaktywne i rezultaty mogą być bez sensu. Prawie na pewno koniec końców coś zrobimy inaczej, niż teraz proponuję, ale trzeba z tym żyć.

Reply:
Na razie zastosowałem sugestie w kodzie, nie ma jeszcze ostrzeżenia.